libs/datamodel/parser-database/src/attributes/default.rs

```diff
@@ -0,0 +1,224 @@
+use crate::{
+    ast::{self, WithName},
+    context::{Arguments, Context},
+    types::{DefaultAttribute, ScalarField, ScalarFieldType, ScalarType},
+    DatamodelError,
+};
+
+/// @default on scalar fields
+pub(super) fn visit_field_default<'ast>(
+    args: &mut Arguments<'ast>,
+    field_data: &mut ScalarField<'ast>,
+    model_id: ast::ModelId,
+    field_id: ast::FieldId,
+    ctx: &mut Context<'ast>,
+) {
+    let value = match args.default_arg("value") {
+        Ok(value) => value,
+        Err(err) => return ctx.push_error(err),
+    };
+
+    let ast_model = &ctx.db.ast[model_id];
+    let ast_field = &ast_model[field_id];
+    if ast_field.arity.is_list() {
+        return ctx.push_error(args.new_attribute_validation_error("Cannot set a default value on list field."));
+    }
+
+    let mapped_name = default_attribute_mapped_name(args, ctx);
+    let default_attribute = args.attribute();
+
+    let mut accept = || {
+        let default_value = DefaultAttribute {
+            value: value.value,
+            mapped_name,
+            default_attribute,
+        };
+
+        field_data.default = Some(default_value);
+    };
+
+    // Resolve the default to a DefaultValue. We must loop in order to
+    // resolve type aliases.
+    let mut r#type = field_data.r#type;
+
+    loop {
+        match r#type {
+            ScalarFieldType::CompositeType(ctid) => {
+                let ct_name = ctx.db.walk_composite_type(ctid).name();
+                ctx.push_error(DatamodelError::new_composite_type_field_validation_error(
+                    "Defaults inside composite types are not supported",
+                    ct_name,
+                    &ast_field.name.name,
+                    args.span(),
+                ));
+            }
+            ScalarFieldType::Enum(enum_id) => {
+                match value.value {
+                    ast::Expression::ConstantValue(enum_value, _) | ast::Expression::BooleanValue(enum_value, _) => {
```

Comment:
is that for enum values true/false?

Reply:
I have an upcoming PR to remove `ast::Expression::BooleanValue`, because if you have an enum with `true` or `false` variants, the expression will be a `BooleanValue` (which is wrong). From a syntax perspective, they are just constant literals. We also get better error messages if we validate that the defaults for boolean values are `true` or `false` later in the process.